src/scheduler/resource_resv.c

```diff
@@ -538,24 +538,29 @@ find_resource_resv_by_indrank(resource_resv **resresv_arr, int rank, int index)
 
 /**
  * @brief
- * 		find a resource_resv by name and start time
+ * 		find a resource_resv by name/index and start time
  *
  * @param[in]	resresv_arr -	array of resource_resvs to search
  * @param[in]	name        -	name of resource_resv to find
  * @param[in]	start_time  -	the start time of the resource_resv
+ * @param[in]	index	    -	index of resource_resv in all_resresv array
+ *				It is ignored when negative
  *
  * @return	resource_resv *
  * @retval	resource_resv	: if found
  * @retval	NULL	: if not found or on error
  *
  */
 resource_resv *
-find_resource_resv_by_time(resource_resv **resresv_arr, char *name, time_t start_time)
+find_resource_resv_by_time_index(resource_resv **resresv_arr, char *name, time_t start_time, int index)
```

Comment:
I don't think you need to change the function's name, it focused on finding resresv by time, not how the list is searched. If you are going to mention index, then you should also mention name.

Reply:
Your comment made me think about it again, I think I should just call a different function in simulation to get the resource_resv object (likely find_resource_resv_by_ind) and not touch this function at all. That is the only place I am passing the index and I think there also I'll retain this call just for standing reservations.